gnat2goto/driver/tree_walk.adb

```diff
@@ -1194,6 +1202,43 @@ package body Tree_Walk is
       return Ret;
    end Do_Constant;
 
+   ------------------------
+   -- Do_String_Constant --
+   ------------------------
+
+   function Do_String_Constant (N : Node_Id) return Irep is
+      Ret              : constant Irep := New_Irep (I_String_Constant_Expr);
+      Element_Type_Ent : constant Entity_Id := Get_Array_Component_Type (N);
+      Element_Type     : constant Irep := Do_Type_Reference (Element_Type_Ent);
+      StrLen           : constant Integer :=
+                                    Integer (String_Length (Strval (N)));
+      String_Length_Expr : constant Irep := New_Irep (I_Constant_Expr);
+   begin
+      --  FIXME: The size of this signedbv should probably not be a hardcoded
+      --         magic number...(e.g. 32 on a 32bit system) this should be set
+      --         programatically some how.
+      Set_Type (String_Length_Expr,
+        Make_Signedbv_Type (Ireps.Empty, 64));
+      --  FIXME: Does this need to be a proper signedbv representation, in hex?
+      Set_Value (String_Length_Expr,
+                 Convert_Uint_To_Hex (UI_From_Int (Int (String_Length
+                   (Strval (N)))), 64));
+
+      Set_Type (Ret,
+                Make_Array_Type (
+                  I_Subtype => Element_Type,
+                  Size => String_Length_Expr));
```

Comment:
This goes in the right direction I think. Right now the main issue is that this won't match the actual type we assign to string literals...

I'm afraid we'll have to do some fairly annoying trickery with the assignment operator to make that work :(